src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -12621,5 +12620,33 @@ void Goo()
 ";
             await VerifyCS.VerifyCodeFixAsync(code, code);
         }
+
+        [Theory, WorkItem(65517, "https://github.com/dotnet/roslyn/issues/65517")]
+        [InlineData("string")]
+        [InlineData("string?")]
```

Comment:
Can you add test where there is `string?` but it affects some other type being created (like a `List<string?>`) through generic type inference. We don't want that inference to now create a different type.

Reply:
Can you give a code sample where removing the cast can mess up things. I cannot come up with something, where removing the cast can create nullability issues

Reply:
```c#    List<string?> GetListOfThings()
    {
        return CreateList((string?)"");
    }

    List<T> CreateList<T>(T value) where T : class? => new List<T> { value };
```

Reply:
Added logic to handle this specific case. However, if I extract the string into a separate variable, something weird happens: https://github.com/dotnet/roslyn/issues/65520. Not sure if it is correct behaviour, so no tests for it so far